lib/y2lan_restart_common.pm

```diff
@@ -148,15 +149,22 @@ sub check_network_status {
         assert_script_run 'dig suse.com|grep \'status: NOERROR\'';    # test if conection and DNS is working
     }
     assert_script_run 'cat journal.log';                              # print journal.log
-    my $is_empty = script_run('[ -s journal.log ]');
-    if ($is_empty) {
-        assert_equals($expected_status, 'no_restart', "Network restart didn't occur, even though it was expected.");
+    my $journal_findings = script_output('cat journal.log');
+    record_info "$expected_status", "Network status expects $expected_status";
+    if ($expected_status eq 'no_restart') {
+        assert_matches(qr/^\s*$/, $journal_findings, "Network service should not have been restarted/reloaded. Check journalctl findings \n$journal_findings\n\n");
+    }
+    elsif ($expected_status eq 'restart') {
+        assert_matches(qr/($query_pattern_for_restart)/, $journal_findings, "Network restart was expected. Check journalctl findings \n$journal_findings\n\n");
+    }
+    elsif ($expected_status eq 'reload') {
+        assert_matches(qr/Reloaded wicked/, $journal_findings, "Network reload was expected. Check journalctl findings \n$journal_findings\n\n");
```

Comment:
In case of reload, we also don't check that there was no restart as it might have happened. But we can leave for future.

Reply:
+1 for implementing now this, IMO to know that strictly reload would be better, I guess we can simply use `assert_not_matches(qr/$query_pattern_for_restart/, $journal_findings` to not change too much the code.

Reply:
i also agree with @jknphy. it seems a trivial change with one or another way

Reply:
added as suggested with assert_not_matches